frontends/p4/tableKeyNames.cpp

```diff
@@ -57,6 +72,9 @@ class KeyNameGenerator : public Inspector {
         name.emplace(expression, l + "[" + r + "]");
     }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("bits")` for
+     * `bits & 0x3` or `0x3 & bits`.
```

Comment:
Key masks are a P4-14 feature only; it is not clear that they are semantically the same as using `&` on a key. In particular, we don't know how the control-plane specifies such keys.